sdk/keyvault/keyvault/src/secrets/index.ts

```diff
@@ -0,0 +1,470 @@
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<DeletedSecret> {
+    const response = await this.client.getDeletedSecret(this.vaultBaseUrl, secretName, options);
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * The purge deleted secret operation removes the secret permanently, without the possibility of
+   * recovery. This operation can only be enabled on a soft-delete enabled vault. This operation
+   * requires the secrets/purge permission.
+   * @summary Permanently deletes the specified secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<void>
+   */
+  public async purgeDeletedSecret(secretName: string, options?: RequestOptionsBase): Promise<void> {
+    await this.client.purgeDeletedSecret(this.vaultBaseUrl, secretName, options);
+  }
+
+  /**
+   * Recovers the deleted secret in the specified vault. This operation can only be performed on a
+   * soft-delete enabled vault. This operation requires the secrets/recover permission.
+   * @summary Recovers the deleted secret to the latest version.
+   * @param secretName The name of the deleted secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async recoverDeletedSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<Secret> {
+    const response = await this.client.recoverDeletedSecret(this.vaultBaseUrl, secretName, options);
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * Requests that a backup of the specified secret be downloaded to the client. All versions of the
+   * secret will be downloaded. This operation requires the secrets/backup permission.
+   * @summary Backs up the specified secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Uint8Array | undefined>
+   */
+  public async backupSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<Uint8Array | undefined> {
+    const response = await this.client.backupSecret(this.vaultBaseUrl, secretName, options);
+    return response.value;
+  }
+
+  /**
+   * Restores a backed up secret, and all its versions, to a vault. This operation requires the
+   * secrets/restore permission.
+   * @summary Restores a backed up secret to a vault.
+   * @param secretBundleBackup The backup blob associated with a secret bundle.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async restoreSecret(
+    secretBundleBackup: Uint8Array,
+    options?: RequestOptionsBase
+  ): Promise<Secret> {
+    const response = await this.client.restoreSecret(
+      this.vaultBaseUrl,
+      secretBundleBackup,
+      options
+    );
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  public async *getSecretVersions(
+    secretName: string,
+    options?: GetAllSecretsOptions
```

Comment:
>options?: GetAllSecretsOptions [](start = 4, length = 30)

I know the options here are the same across the GetAll* operations for secrets, but does it still make sense to have distinct options types for each of the operations?  I don't have a strong opinion and would defer to people with more TS experience.

Reply:
We can optionally fold them into one if feedback is that it's confusing to have them separate.

Reply:
In coordination with @jonathandturner, let's address this through this issue: https://github.com/Azure/azure-sdk-for-js/issues/3030